rust/datafusion/src/sql/planner.rs

```diff
@@ -638,10 +638,13 @@ impl<'a, S: SchemaProvider> SqlToRel<'a, S> {
                 Ok(Expr::IsNotNull(Box::new(self.sql_to_rex(expr, schema)?)))
             }
 
-            SQLExpr::UnaryOp { ref op, ref expr } => match *op {
-                UnaryOperator::Not => {
+            SQLExpr::UnaryOp { ref op, ref expr } => match (op, expr) {
+                (UnaryOperator::Not, _) => {
                     Ok(Expr::Not(Box::new(self.sql_to_rex(expr, schema)?)))
-                }
+                },
+                (UnaryOperator::Minus, box SQLExpr::Value(Value::Number(n))) =>
+                    // Parse negative numbers properly
+                    Ok(lit(-n.parse::<f64>().unwrap())),
```

Comment:
`n` could be an i64 or f64. You could maybe recursively call sql_to_rex to parse the `Value::Number` here.

Reply:
Ok I'll parse both i64 and f64; however I can't recursively call `sql_to_rex` as I need to negate the number.